swarm/src/connection.rs

```diff
@@ -454,15 +470,6 @@ where
     }
 }
 
-fn gather_supported_protocols(handler: &impl ConnectionHandler) -> HashSet<StreamProtocol> {
-    handler
-        .listen_protocol()
-        .upgrade()
-        .protocol_info()
-        .filter_map(|i| StreamProtocol::try_from_owned(i.as_ref().to_owned()).ok())
-        .collect()
-}
```

Comment:
I think it would be worth restoring this function but modified to return `handler.listen_protocol().upgrade().protocol_info()`.